content/new-routing.md

```diff
@@ -0,0 +1,446 @@
+
+```js
+FlowRouter.route('/old-list-route/:_id', {
+  action(params) {
+    FlowRouter.go('listsShow', params);
+  }
+});
+```
+
+### Redirecting dynamically
+
+If however, you need some data to redirect, you'll need to render part of the component hierarchy, as that is the place where data subscribing happens. For example, in the Todos example app, we want to make the root (`/`) route redirect to the first known list. To achieve this, we need to render a special `rootRedirector` route:
+
+```js
+FlowRouter.route('/', {
+  name: 'home',
+  action: () => {
+    BlazeLayout.render('appBody', {main: 'rootRedirector'});
+  }
+});
+```
+
+Because the `rootRedirector` template is rendered inside the `appBody` layout which takes care of subscribing to the set of lists the user knows about *before* rendering it's sub-template, and we are guaranteed there is at least one such list, we can simply do:
+
+```js
+Template.rootRedirector.onCreated(() => {
+  // We need to set a timeout here so that we don't redirect from inside a redirection
+  //   which is a no-no in FR.
+  Meteor.setTimeout(() => {
+    FlowRouter.go('listsShow', Lists.findOne());
+  });
+});
+```
+
+### Redirecting after a user's action
+
+Often, you just want to go to a new route programmatically when a user has completed a certain action. Above we saw a case (creating a new list) when we wanted to do it *optimistically*---i.e. before we hear back from the server that the Method succeeded. We can do this because we reasonably expect that the Method will succeed in almost all cases (see the {% link_to 'ui-ux' 'UI/UX article'} for further discussion of this).
+
+However, if we wanted to wait for the method to return for the server, we can put the redirection in the callback of the method:
+
+```js
+Template.appBody.events({
+  'click .js-new-list'() {
+    Lists.methods.insert.call((err, listId) => {
+      if (!err) {
+        FlowRouter.go('listsShow', { _id: listId });  
+      }
+    });
+  }
+});
+```
+
+You will also want to show some kind of status while the method is working so that the user knows there is something going on between them clicking the button and the redirect happening (and show the error some kind of message if the error is there too).
+
+## Advanced Routing
+
+### Missing pages
+
+If a user types an incorrect URL, chances are you want to show them some kind of amusing not found page. There are actually two categories of "not found" pages. The first is when the URL typed in doesn't match any of your route definitions. You can use `FlowRouter.notFound()` to handle this:
+
+```js
+// the appNotFound template is used for unknown routes and missing lists
+FlowRouter.notFound = {
+  action() {
+    BlazeLayout.render('appBody', {main: 'appNotFound'});
+  }
+};
+```
+
+The second is when the URL is valid, but doesn't actually match any data. Here we
```

Comment:
Oh yeah, I mean there's a code todo there, I'll finish the sentence once that's done :)